app/schema.graphql

```diff
@@ -0,0 +1,103 @@
+scalar Datetime
+
+type Author {
+  name: String
+  email: String
+}
+
+# git commit data.
+type Commit {
+  sha1: String!
+  author: Author!
+  date: Datetime!
+  message: String!
+  refs: [Ref]
+  objectName: ObjectName!
+}
+
+# file info and metadata.
+type Info {
+  # base name of the file.
+  name: String!
+  # length in bytes.
+  size: Int!
+  # if it's a directory.
+  isdirectory: Boolean!
+  # any flags git stores, currently only executable.
+  flags: [Int]!
```

Comment:
What's this?

Reply:
Mode flags, currently the only one git preserves afaik is the executable flag on a file.

Reply:
Ah, ya `Blob` has this https://docs.rs/git2/0.10.1/git2/struct.Blob.html#method.is_binary. I think Pijul actually has the whole set of modes :)

Reply:
Not sure if this is the same, as git supports this: `git update-index --chmod=+x foo.sh`. In general executbales dont need to be binary.

Reply:
Would it make sense to use `mode` similar to what `git-tree` returns?
```
mvp/app » git ls-tree --long code-browsing /Users/rudolfs/work/mvp/
040000 tree e8023e1209a6022568f04962f0cdf48f815b48d3       -	../.buildkite
100644 blob 3c3629e647f5ddf82548912e337bea9826b434af      13	../.gitignore
040000 tree fe3470499dcdf36bc6c048c8d43cae67554e27db       -	./
040000 tree 57f6fc0dced5a21b9af6721340920c3f0eae899d       -	../proxy
```

Reply:
I wonder where this information is stored then? Basically need to figure out if you can get it from the `Commit` or will I have to store some information alongside `File` and `Directory` in the `surf` model.

Reply:
Oh this is probably it: https://docs.rs/git2/0.10.1/git2/struct.TreeEntry.html#method.filemode